frontend/src/components/common/form/TimePicker.tsx

```diff
@@ -0,0 +1,216 @@
+import type { ForwardedRef, ReactElement } from "react";
```

Comment:
what do you think abt center aligning instead of aligning ends

<img width="134" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/e11efdf1-2f44-474c-bcbc-2b89a28fee1b">

<img width="136" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/e9094db1-f1be-4242-9ea1-6a8209e59603">

and decreasing padding at the ends from 4->2
<img width="113" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/256042e4-b014-43f5-8651-21f4ff86eeba">

Reply:
also idk if we necessarily need the blue background. i feel like the cursor could be clear enough. thoughts?
<img width="152" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/7481461c-edb7-4a15-b0ae-75035adcfcf7">

Reply:
I have a strong preference for end alignment because (1) the cursor is placed in a more logical location, and (2) it looks more like a timestamp that you've entered in text, i.e., `03:05` instead of `03 : 05`. Let me know if you see a good reason otherwise, though.

In terms of the blue background, I included it because this matches the styling of other time pickers, including a library from Adobe and the built-in picker in Chrome. I'm not opposed to changing it, though I do like how the blue background is an indicator on which field is focused (aside from the caret, which might be hard to see). What do you think?

Reply:
we can keep as is